plugins/rest-api.php

```diff
@@ -0,0 +1,422 @@
+
+	return compact( 'changed', 'status' );
+}
+
+/**
+ * Determine if the current request is for a REST API endpoint
+ *
+ * Normally we would check REST_REQUEST for this, but that doesn't get set until `parse_query`
+ * runs during phase 2, so instead, we need to pull the API prefix in `wp-cache-config.php`
+ * and compare the URI ourselves.
+ *
+ * @return bool
+ */
+function wpsc_rest_is_api_request() {
+	global $wp_cache_request_uri, $wp_cache_rest_prefix;
+	$is_request = false;
+
+	if ( substr( $wp_cache_request_uri, 1, strlen( $wp_cache_rest_prefix ) ) === $wp_cache_rest_prefix ) {
+		$is_request = true;
+	}
+
+	return $is_request;
+}
+
+/*
+ * Always try to serve supercache files for REST API requests
+ *
+ * `$wp_cache_slash_check` is automatically set to `true` by `wp_cache_manager()` when the
+ * permalink structure has a trailing slash. When looking for an existing cache file,
+ * `wp_cache_serve_cache_file()` assumes files will either have a trailing slash or won't,
+ * and doesn't handle the case where both forms are valid.
+ *
+ * Both forms are valid for REST API endpoints, though, so we need to tell
+ * `wp_cache_serve_cache_file()` to look for a supercache file.
+ */
+function wpsc_rest_serve_supercache_file( $serve_supercache_file ) {
+	if ( wpsc_rest_is_api_request() ) {
+		$serve_supercache_file = true;
+	}
+
+	return $serve_supercache_file;
+}
+
+/**
+ * Send the correct headers for JSON output
+ *
+ * By default WPSC sends `Content-Type: text/html`, which could cause the browser to interpret
+ * the output as HTML, opening up XSS attack vectors.
+ */
+function wpsc_send_json_headers() {
+	if ( ! wpsc_rest_is_api_request() ) {
+		return;
+	}
+
+	// The JSON specification says that JSON is always Unicode, and no charset is needed
+	header( 'Content-type: application/json' );
+}
+
+/**
+ * Save the REST API URL prefix in WPSC's config
+ *
+ * This is necessary for `wpsc_rest_is_api_request()` to work. See the comments there for details.
+ */
+function wpsc_rest_save_url_prefix() {
+	global $wp_cache_rest_prefix, $wp_cache_config_file;
+
+	$canonical_prefix = rest_get_url_prefix();
+
+	if ( $canonical_prefix === $wp_cache_rest_prefix ) {
+		return;
+	}
+
+	// Update global variable for use later in this request
+	$wp_cache_rest_prefix = $canonical_prefix;
+
+	// Update stored configuration
+	wp_cache_replace_line(
+		'^ *\$wp_cache_rest_prefix',
+		"\$wp_cache_rest_prefix = '$canonical_prefix';",
+		$wp_cache_config_file
+	);
+}
+
+/**
+ * Register JSON end-of-file markers
+ *
+ * `wp_cache_get_ob()` uses end-of-file markers to determine which types of pages to cache.
+ * Adding JSON markers causes WPSC to cache files ending with JSON markers.
+ *
+ * @param string $eof_pattern
+ *
+ * @return string
+ */
+function wpsc_rest_eof_tags( $eof_pattern ) {
+	$json_object_pattern     = '^[{].*[}]$';
+	$json_collection_pattern = '^[\[].*[\]]$';
```

Comment:
Note that `null`, `"hello"`, and `21345` are all completely valid JSON responses as well, but core wouldn't generate that by default.